flutter-mentorship-t4/mentorship_project
# Feature/categories/diva 21 categories screen UI logic

# Pull Request: Categories Logic Implementation

## Jira Ticket: [DIVA-21](https://nz-dev.atlassian.net/jira/software/projects/DIVA/boards/1?selectedIssue=DIVA-21)

## Description

I have completed the implementation of the logic for the **Categories Screen** in our Flutter app. This PR includes state management for loading categories, handling errors, and navigating to category details.

## How to Test

1. Open the Flutter app.
2. Navigate to the **Categories Screen**.
3. Verify that:
   * Categories load correctly with a loading indicator during data retrieval.
   * If the API request fails, an error message is displayed.
4. Select a category:
   * Verify that the app navigates to the **Category Detail** screen.
   * Ensure that the selected category’s data is displayed correctly.
5. Refresh the **Categories Screen** to ensure that data is fetched again and the state updates correctly.
6. Test edge cases such as slow network conditions and verify that the loading indicator displays until data is loaded.

## Screenshots
<img src="https://github.com/user-attachments/assets/08e8913f-a268-44f9-bfa0-b01d22f374f6" width="200"/>
<img src="https://github.com/user-attachments/assets/08e8913f-a268-44f9-bfa0-b01d22f374f6" width="200"/>
<img src="https://github.com/user-attachments/assets/1f92640b-8c01-4734-8191-5e02200dc650" width="200"/>
<img src="https://github.com/user-attachments/assets/e6961ac4-4a27-46af-838f-372ec38a6f49" width="200"/>



[DIVA-21]: https://nz-dev.atlassian.net/browse/DIVA-21?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
